Handle oneOf keywords

xela85 opened the pull request:
Thanks to https://github.com/eikek/sbt-openapi-schema/pull/8, sbt-openapi is able to parse `allOf` declarations and transform them in `sealed trait` hierarchies. 

However, it is still not able to handle other keywords such as `anyOf` or `not` or `oneOf` (https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/). Hence, a `NullPointerException` is thrown while trying to do so as the plugin always tries to get the `allOf` value of a `ComposedSchema`.

This pull request consists in handling the `oneOf` keywords by converting it in a `allOf` definition after having parsed the swagger file.

eikek commented:
Thank you so much!  For my understanding:  `oneOf` and `allOf` are handled "the same" is that correct? And this is – as before with `allOf` – scala-only, right?

if you don't mind, could you add only a few words about this to the README? doesn't need to be huge text, just  so people know that it exists (and optionally a little how it works). If not possible, that's also fine!

xela85 commented:
I have updated the README, I hope that the example makes it clearer. In my opinion a `allOf` looks more like inheritance : all the fields defined in the parent (referenced using `allOf`) are implemented in children. In `oneOf` that is the parent which defines its children. Therefore it seems not possible for the parent to expose common fields of its children in an object oriented way (`abstract` vals). 

In Scala `allOf` without shared values and the opposed `oneOf` lead to the same generated code, that's why I implemented a preprocessing task after the parsing which inverts `oneOf` relations. 

It may be better to refactor types used in parsing and grouping, but I decided to stay with the current ones in order to minimize diffs.

eikek commented:
I see now, thank you very much - this is very nice, also thanks for the readme update; it makes it very clear now for me.
